sdk/storage/AzureStorageBlob/Source/Operations/BlockOperation.swift

```diff
@@ -84,8 +84,10 @@ internal class BlockOperation: TransferOperation {
                     self.notifyDelegate(withTransfer: parent)
                 case let .failure(error):
                     var err = error
-                    if let pipelineError = error as? PipelineError {
-                        err = pipelineError.innerError
+                    switch error {
+                    case let .wrapped(innerError, _):
+                        err = AzureError.service(innerError.localizedDescription)
```

Comment:
I'm not sure if this error is actually one that is returned from the service. IIRC "service" is intended to only be for deserialized error response objects actually returned back by the service in its response.

Reply:
Updated.